### PR TITLE
[TRIGGER] PayPal #6627

### PR DESCRIPTION
**Pull request by lcaresia**
… events

### <samp>🤖 Generated by Copilot at c30379d</samp>

This pull request adds a new Paypal app component and a new source component for emitting new payment events. It also updates the `@pipedream/platform` module to the latest version and removes some unused files. The new app component defines the app props and methods for interacting with the Paypal API, and the new source component inherits from a common module that handles webhook creation, deletion, and event processing. The pull request also updates the `pnpm-lock.yaml` file to reflect the dependency changes.

### <samp>🤖 Generated by Copilot at c30379d</samp>

> _The Paypal app got a new action_
> _To get user profile information_
> _And a source for payment events_
> _With webhooks and common sense_
> _And some updates for the platform version_

### <samp>🤖 Generated by Copilot at c30379d</samp>

*  Add a new app component for the Paypal app that defines the app type, prop definitions, and methods for interacting with the Paypal API ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-40d7e994f30e75b8ae012c1ae340338e106db6db121a98bb7f34eb518667fac4R1-R76))
* Add a new action component for the Paypal app that allows users to get user profile information from the Paypal API ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-e4d7fd86087b5503f4740faa53cd4bc02a28e95876491d2e0d1a3d5f681ead48R1-R23))
* Add a new source component for the Paypal app that emits new events on new payment webhook events ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-abf3c24512101e76124158d67827b42890352c7ae8bb4b3701281995528b4296R1-R40))
* Add a new common module for the Paypal source components that defines common props, methods, and hooks for creating and removing webhooks, getting webhook event types, emitting events, and handling webhook events ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-6f6ab11c893a62c3b036fdbaa8e25945ec4aa94b919df0c41684a400f25022f7R1-R50))
* Add a new package.json file for the Paypal app component that specifies the name, version, description, main, keywords, homepage, author, gitHead, publishConfig, and dependencies of the component ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-7486dc028f2fb6314a08cf42c83ac85c31e2486b2a714f291e0c73a992efb059R1-R19))
* Update the dependency version of the `@pipedream/platform` module from 1.1.1 to 1.5.1 in the package.json file of the root directory and the pnpm-lock.yaml file to ensure compatibility with the new Paypal app component that uses the axios module from the platform ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L66-R66), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8-R8))
* Add new entries for the callrail, chatwork, fastfield_mobile_forms, iauditor_by_safetyculture, trakt, and wave app components in the pnpm-lock.yaml file as part of separate pull requests that add new app components for these apps ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR619-R621), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR676-R678), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1361-R1363), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2026-R2028), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4673-R4675), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4999-R5001))
* Add a new entry for the paypal app component in the pnpm-lock.yaml file that specifies the specifiers and dependencies of the component ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3108-R3113))
* Update the dependency version of the `@types/node` module from 17.0.45 to 18.15.12 in the pnpm-lock.yaml file to ensure compatibility with the latest Node.js version and the typescript compiler ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9119-R9143), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9202-R9226), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9314-R9338), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9324-R9348), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9351-R9375), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9363-R9387), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9404-R9428), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9584-R9608), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11836-R11860), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11861-R11885), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11868-R11892), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11874-R11898), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11922-R11946), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12064-R12088), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12085-R12109), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17595-R17619), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17623-R17647), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17881-R17905), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17899-R17923), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17911-R17935), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17932-R17956), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17973-R17997), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18061-R18085), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18069-R18093), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18168-R18192), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18284-R18308), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18403-R18427), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18428-R18452), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18437-R18461), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21603-R21627))
* Delete the unused files `components/paypal/actions/userinfo/userinfo.mjs` and `components/paypal/paypal.app.mjs` ([link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-6c83b99a76820ba643c9d1ef572990ae50970fc5808d13f6e51f970e5addd7a5), [link](https://github.com/PipedreamHQ/pipedream/pull/6639/files?diff=unified&w=0#diff-86fa345b1dca5b463db7e0a8b0333e0b473038477de96faa74eefc2f240ec713))

### CODE REVIEW

File: components/paypal/actions/get_user_info/get_user_info.mjs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import app from "../../app/paypal.app.mjs";
+
+export default {
+  key: "paypal-get-user-info",
+  name: "Get User Info",
+  description: "Shows user profile information. [See docs here](https://developer.paypal.com/docs/api/identity/v1)",
```
Comment (dylburger) [inline components/paypal/actions/get_user_info/get_user_info.mjs:6]:
```suggestion
  description: "Displays user profile information. [See docs here](https://developer.paypal.com/docs/api/identity/v1)"
```

Comment (dylburger) [reply components/paypal/actions/get_user_info/get_user_info.mjs:6]: Changed 'Shows' to 'Displays' to follow Google's Developer docs style guide (Use present tense and active voice)

File: components/paypal/sources/new-payment-event/new-payment-event.mjs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import common from "../common/common.mjs";
+
+export default {
+  ...common,
+  name: "New Payment Event (Instant)",
+  version: "0.0.1",
+  key: "paypal-new-payment-event",
+  description: "Emit new event on an new payment webhook event. [See webhooks docs here](https://developer.paypal.com/docs/api/webhooks/v1/#webhooks_post) and [Events Types docs here](https://developer.paypal.com/docs/api/webhooks/v1/#webhooks-event-types_list)",
```
Comment (dylburger) [inline components/paypal/sources/new-payment-event/new-payment-event.mjs:8]:
```suggestion
  description: "Emits a new event on a new payment webhook event. [See webhooks docs here](https://developer.paypal.com/docs/api/webhooks/v1/#webhooks_post) and [Events Types docs here](https://developer.paypal.com/docs/api/webhooks/v1/#webhooks-event-types_list)"
```

Comment (dylburger) [reply components/paypal/sources/new-payment-event/new-payment-event.mjs:8]: Changed 'Emit new event' to 'Emits a new event' to follow Google's Developer docs style guide (Use present tense and active voice)

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Paypal_6627_1555.pdf](https://github.com/PipedreamHQ/pipedream/files/11573365/Paypal_6627_1555.pdf)

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**